### PR TITLE
Fix regression when reading and deleting text in MS Excel edit fields.

### LINKED ISSUE #9042: NVDA fails to announce cursor movement and character deletion in go to edit field in excel 2010
### Steps to reproduce:
1. Start excel 2010.
2. Press CTRL+g
3. Type b2:b11.
4. try to move with arrow keys or deleting with backspace.
### Actual behavior:
Blank is spoken.
### Expected behavior:
Deleted characters should be announced and characters to which cared moves should be spoken.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
NVDA version alpha-16405,19bb21c2

#### Windows version:
Windows 7 x64
#### Name and version of other software in use when reproducing the issue:
Microsoft  Excel 2010  32-bit
#### Other information about your system:
The developer info for this field is as follows.
Developer info for navigator object:
name: u'Odwo\u0142anie:'
role: ROLE_EDITABLETEXT
states: STATE_FOCUSABLE, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <baseObject.Dynamic_IAccessibleUnidentifiedEditWindowNVDAObject object at 0x056734F0>
Python class mro: (<class 'baseObject.Dynamic_IAccessibleUnidentifiedEditWindowNVDAObject'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.edit.UnidentifiedEdit'>, <class 'NVDAObjects.window.edit.RichEdit'>, <class 'NVDAObjects.window.edit.Edit'>, <class 'NVDAObjects.behaviors.EditableTextWithAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: RectLTWH(left=476, top=394, width=261, height=14)
value: None
appModule: <'appModuleHandler' (appName u'excel', process ID 5196) at address 5261370>
appModule.productName: u'Microsoft Office 2010'
appModule.productVersion: u'14.0.7224.5000'
TextInfo: <class 'NVDAObjects.window.edit.EditTextInfo'>
windowHandle: 4588610
windowClassName: u'EDTBX'
windowControlID: 1027
windowStyle: 1409286272
windowThreadID: 5624
windowText: u'b2:b11'
displayText: u'b2:b11'
IAccessibleObject: <POINTER(IAccessible) ptr=0x70bd060 at 548eda0>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=4588610, objectID=-4, childID=0
IAccessible accName: u'b2:b11'
IAccessible accRole: ROLE_SYSTEM_CLIENT
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048580)
IAccessible accDescription: None
IAccessible accValue: None

#### Does the issue still occur after restarting your PC?
Yes
#### Have you tried any other versions of NVDA?
Yes it is a regression. This is working correctly in NVDA 2018.1 and failing in NVDA 2018.2 and above
cc @michaelDCurran @feerrenrut This is pretty serious for those using Excel 2010, because it fails similarly in many more edit fields. Can this be assigned P2 ad least and fixed asap?

### PR DESCRIPTION
**Pull request by lukaszgo1**
## Link to issue number:
Fixes #9042
### Summary of the issue:
When #8165 was introduced it was assumed, that if particular edit field exposes line information it do not have to be accessed via displayModel. It unfortunately broke accessibility of most edit fields in Excel.
### Description of how this pull request fixes the issue:
For the class used by these fields displayModel is used again.
### Testing performed:
Tested with go to, find and custom list editing in Microsoft Excel 2010 and 2019.
### Known issues with pull request:
I am pretty new to this part of the code, so it is possible, that it should be solved in a completely different way. @leonardder your opinion would be appreciated.
### Change log entry:

Section: Bug fixes

NVDA will again track cursor and announce deleted characters in edit fields in Microsoft Excel such as go to or find.

### CODE REVIEW

**Review by LeonarddeR — CHANGES_REQUESTED** [review]
I think we should do either one of the following.

1. Create a list of badUdentifiedEditClassNames and list this window class in there.
2. If this window class is specific to Excel, which I think it is, this should be handled in an Excel appModule
3. Similar to the windowText and windowTextLineCount properties, create a windowTextLength property that fetches the window text length:
    `return watchdog.cancellableSendMessage(nav.windowHandle,winUser.WM_GETTEXTLENGTH,0,0)`
    Then, only set UnidentifiedEdit as an overlay if there is text in the control. For Excel, this returns 0 even though there is text. A drawback of this approach is that unidentified text edits will always initialize with a display model overlay class when they are found to be empty.

This is clearly a bad implementation in Excel, as the text is perfectly provided using the window text.

@MichaeldCurran: Do you have thoughts about this? I prefer either 2 or 3.

**Comment by lukaszgo1** [comment]
For me option two looked most reasonable, so I've committed required changes. Of course if @michaelDCurran would have a different opinion I am ready to reorganize it.

File: source/appModules/excel.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+import appModuleHandler
+from NVDAObjects.window import DisplayModelEditableText
+
+class AppModule(appModuleHandler.AppModule):
+
+	def chooseNVDAObjectOverlayClasses(self, obj, clsList):
+		windowClass = obj.windowClassName
+		# #9042 Edit fields in Excel have to be accessed via displayModel.
```
Comment (leonardder) [inline source/appModules/excel.py]: There usually is a colon after the issue number in such comments.

```suggestion
 		# #9042: Edit fields in Excel have to be accessed via displayModel.
```

File: source/appModules/excel.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+		if windowClass == "EDTBX":
+			clsList.insert(0, DisplayModelEditableText)
```
Comment (leonardder) [inline source/appModules/excel.py]: You will also have to import UnidentifiedEdit from NVDAObjects.window.edit and remove that from the clsList. Make sure to wrap this in a try/except statement. See the putty module as an example.

**Review by LeonarddeR — CHANGES_REQUESTED** [review]
I think this is the way to go indeed.

**Comment by lukaszgo1** [comment]
@leonardder All your commends are now addressed.

**Review by LeonarddeR — APPROVED** [review]

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**